gammapy/utils/fitting/tests/test_datasets.py

```diff
@@ -0,0 +1,25 @@
+import pytest
+import numpy as np
+from numpy.testing import assert_allclose
+from .test_fit import MyDataset
```

Comment:
Should the test rely only on a fake `Dataset` or should the be specific tests for each type of dataset and also for a heterogeneous set? Maybe this is for later.

Reply:
I'll add a test for multiple `MapDataset` now, the joint fit for `SpectrumDataset` and `FluxPointsDataset` could come later...